src/main/java/org/elasticsearch/index/query/BaseTermQueryBuilder.java

```diff
@@ -0,0 +1,242 @@
+        // for serialization only
+    }
+
+    /** Returns the field name used in this query @see org.apache.lucene.search.spans.SpanTermQuery#getField() */
+    public String fieldName() {
+        return this.fieldName;
+    }
+
+    /** Returns the value used in this query @see org.apache.lucene.index.Term#text() */
+    public Object value() {
+        return this.value;
+    }
+    
+    /** Returns the query name for the query. */
+    public String queryName() {
+        return this.queryName;
+    }
+    /**
+     * Sets the query name for the query.
+     */
+    @SuppressWarnings("unchecked")
+    public QB queryName(String queryName) {
+        this.queryName = queryName;
+        return (QB) this;
+    }
+    
+    /** Returns the boost for this query. @see org.apache.lucene.search.Query#setBoost() */
+    public float boost() {
+        return this.boost;
+    }
+    /**
+     * Sets the boost for this query.  Documents matching this query will (in addition to the normal
+     * weightings) have their score multiplied by the boost provided.
+     */
+    @SuppressWarnings("unchecked")
+    @Override
+    public QB boost(float boost) {
+        this.boost = boost;
+        return (QB) this;
+    }
+
+    @Override
+    protected void doXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject(parserName());
+        Object valueToWrite = value;
+        if (value instanceof BytesRef) {
+            valueToWrite = ((BytesRef) value).utf8ToString();
+        }
+
+        if (boost == 1.0f && queryName == null) {
+            builder.field(fieldName, valueToWrite);
+        } else {
+            builder.startObject(fieldName);
+            builder.field("value", valueToWrite);
+            if (boost != 1.0f) {
+                builder.field("boost", boost);
+            }
+            if (queryName != null) {
+                builder.field("_name", queryName);
+            }
+            builder.endObject();
+        }
+        builder.endObject();
+    }
+    
+    /** Returns a {@link QueryValidationException} if fieldName is null or empty, or if value is null. */
+    @Override
+    public QueryValidationException validate() {
+        QueryValidationException validationException = null;
+        if (fieldName == null || fieldName.isEmpty()) {
+            validationException = QueryValidationException.addValidationError("field name cannot be null or empty.", validationException);
+        }
+        if (value == null) {
+            validationException = QueryValidationException.addValidationError("value cannot be null.", validationException);
+        }
+        return validationException;        
+    }
+
+    
+    @Override
+    public int hashCode() {
+        /* 
+         * Added class name to the hashcode so a {@link SpanTermQueryBuilder} and a {@link TermQueryBuilder} aren't considered the same
+         * if they have the same parameters set.
+         */
```

Comment:
this makes sense, wonder if the comment is even needed, seems more like a commit message ;)